client/electron/vpn_service.ts

```diff
@@ -88,3 +82,68 @@ let statusCb: VpnStatusCallback | undefined = undefined;
 export function onVpnStatusChanged(cb: VpnStatusCallback): void {
   statusCb = cb;
 }
+
+/**
+ * Forwards the VPN connection status change event to the VpnStatusCallback.
+ * @param connJson The JSON string representing the VPNConn interface.
+ */
+function handleVpnConnectionStatusChanged(connJson: string) {
+  const conn = JSON.parse(connJson) as VPNConn;
+  console.debug(`received ${StatusChangedEvent}`, conn);
+  switch (conn?.status) {
+    case VPNConnConnected:
+      statusCb?.(conn.id, TunnelStatus.CONNECTED);
+      break;
+    case VPNConnConnecting:
+      statusCb?.(conn.id, TunnelStatus.RECONNECTING);
+      break;
+    case VPNConnDisconnecting:
+      statusCb?.(conn.id, TunnelStatus.DISCONNECTING);
+      break;
+    case VPNConnDisconnected:
+      statusCb?.(conn.id, TunnelStatus.DISCONNECTED);
+      break;
+  }
+}
+
+// Callback token of the VPNConnStatusChanged event
+let vpnConnStatusChangedCb: CallbackToken | undefined;
+
+/**
+ * Subscribes to all VPN connection related events.
+ * This function ensures that the subscription only happens once.
+ */
+async function subscribeVPNConnEvents(): Promise<void> {
+  if (!vpnConnStatusChangedCb) {
+    vpnConnStatusChangedCb = await newCallback(
+      handleVpnConnectionStatusChanged
+    );
+    await invokeGoMethod(
+      'AddEventListener',
+      JSON.stringify({
+        name: StatusChangedEvent,
+        callbackToken: vpnConnStatusChangedCb,
+      })
+    );
+  }
+}
+
+//#region type definitions of VPNConnection in Go
+
+// The following constants and types should be aligned with the corresponding definitions
+// in `./client/go/outline/vpn/vpn.go`.
+
+const StatusChangedEvent = 'VPNConnStatusChanged';
+
+type VPNConnStatus = string;
+const VPNConnConnecting: VPNConnStatus = 'Connecting';
+const VPNConnConnected: VPNConnStatus = 'Connected';
+const VPNConnDisconnecting: VPNConnStatus = 'Disconnecting';
+const VPNConnDisconnected: VPNConnStatus = 'Disconnected';
+
+interface VPNConn {
```

Comment:
This is not a VPN connection: you can't read/write to it.

Perhaps call it `VPNStatus`?

Reply:
`VPNConnectionStatus` or `VPNConnectionState` sounds good to me.